src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1356,10 +1356,24 @@ internal HttpRequestMessage CreateRequest<TResult>(JsonOperationContext ctx, Ser
 
                 case HttpStatusCode.Forbidden:
                     var msg = await TryGetResponseOfError(response).ConfigureAwait(false);
-                    throw new AuthorizationException("Forbidden access to " + chosenNode.Database + "@" + chosenNode.Url + ", " +
-                        (Certificate == null ? "a certificate is required. " : Certificate.FriendlyName + " does not have permission to access it or is unknown. ") +
-                        $"Method: {request.Method}, Request: {request.RequestUri}" + Environment.NewLine + msg
-                        );
+                    var builder = new StringBuilder(msg)
+                        .Append(". ").AppendLine()
+                        .Append("Forbidden access to ").Append(chosenNode.Database).Append("@").Append(chosenNode.Url).Append(", ");
+                    if (Certificate == null)
+                    {
+                        builder.Append("a certificate is required. ");
+                    }
+                    else if(Certificate.PrivateKey != null)
+                    {
+                        builder.Append(Certificate.FriendlyName).Append(" does not have permission to access it or is unknown. ");
+                    }
+                    else
+                    {
+                        builder.Append("The certificate ").Append(Certificate.FriendlyName)
```

Comment:
This is the part where we tell you that the provided certificate is missing the private key, which is what is important, no?

Reply:
Append(" contains no private key. Constructing the certificate with the 'X509KeyStorageFlags.MachineKeySet' flag may solve this problem. "); 
https://github.com/ravendb/ravendb/pull/11452/files#diff-8eb5dec408a858236268c82f3fe1523a8d957370e6a24bef4e21079f70d36ae1R1373